src/modes/shop/shop_utils.cpp

```diff
@@ -61,7 +62,15 @@ ShopObject::ShopObject(GlobalObject *object) :
     _trade_price = _object->GetTradingPrice();
 }
 
-
+ShopObject::~ShopObject()
+{
+    // Do not delete global objects owned by the inventory system.
+    if (!_is_inventory_item &&
```

Comment:
ShopObjects were not cleaning up their GlobalObjects.  So, I added code to do this.

However, not all of these GlobalObjects are owned by the shop itself.  Some items are owned by the party's inventory system and added to the shop to sell.  The shop is not responsible for freeing these items.

I added a boolean flag to facilitate this.  I do not particularly like this solution.  However, it does solve the problem at the moment.

I think there is a more elegant solution to handle this.